### PR TITLE
[Clamp] Add a proposal

### PR DESCRIPTION
**Pull request by nex3**
See #2860

### CODE REVIEW

**Comment by nex3** [comment]
@mirisuzanne True confession: I forgot https://github.com/sass/sass/pull/2863 existed until I had finished writing this proposal :p. If you'd rather revise yours, please feel free to go ahead, but if not could you give mine a review?

File: proposal/clamp.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+doesn't seem too widespread yet judging by the lack of support requests and only
+two 👍s on the issue in the issue tracker.
+
+[caniuse]: https://caniuse.com/mdn-css_types_clamp
+
+## Summary
+
+This proposal makes `clamp()` essentially a synonym of `calc()`, so that its
+contents are parsed in the same liberal manner with interpolation as the only
+valid Sass use of Sass within them.
```
Comment (mirisuzanne) [inline proposal/clamp.md]:
```suggestion
valid use of Sass within them.
```

File: proposal/clamp.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+### Special Number String
+
+`clamp(` is added to the list of possible prefixes for a [special number
+string].
+
+[special number string]: ../spec/functions.md#special-number-string
+
+### Procedures
+
+## Syntax
```
Comment (mirisuzanne) [inline proposal/clamp.md]:
```suggestion
## Syntax
```

**Review by mirisuzanne — APPROVED** [review]
looks good to me - with a few minor typos, I think. I'm not precious about my PR, so feel free to close it. Glad this is moving forward!

File: proposal/clamp.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+variables. This does represent a potential backwards-incompatibility for users
+who have started using `clamp()` with Sass's default function syntax, which
+interprets all arguments as SassScript expressions. However, outside of obvious
+cases like a single variable being used as an argument, it's difficult to
+disambiguate Sass expressions and plain-CSS math expressions, so we'd like to
+avoid needing to do so if at all possible.
+
+It's worth noting that there is prior art for this disambiguation. When adding
+support for [plain-CSS `min()` and `max()`] functions, we decided to
+disambiguate the pain CSS versions from the Sass-syntax versions by first
```
Comment (jathak) [inline proposal/clamp.md]:
```suggestion
disambiguate the plain CSS versions from the Sass-syntax versions by first
```

**Review by jathak — APPROVED** [review]

**Pull request merged**